### DISCUSSION TITLE
Replace idaapi.read_selection() with ida_kernwin.read_range_selection()

### DISCUSSION THREAD

**Discussion by Harding-Stardust** [arizvisa/ida-minsc#184] [Ideas]
Replace idaapi.read_selection with ida_kernwin.read_range_selection. Since is a much nicer API to use.
```
is_valid_selection, sel_start, sel_end = ida_kernwin.read_range_selection(None)
```

https://github.com/arizvisa/ida-minsc/blob/af07f167a0992aed3f3a2a1670593880baa57820/misc/ui.py#L304

**Comment by arizvisa** [comment]
Isn't the `ui.current.selection` function only returning the left and right boundaries? I'm not sure what else that `read_range_selection` exposes that makes it different from `read_selection`. Can you clarify?

Also, using `ui.current.selection` directly is entirely unnecessary. You **should** be able to get the exact same functionality using the `database.address` namespace since you're essentialy looking for an address range.

Example:
```
ui.current.address() == database.address()
ui.current.selection() == database.address.bounds()       # make sure you select a range
```
In all actuality, you should be able to use `db.address` for nearly everything.
If you want a list of addresses that are currently selected, the following two things are equivalent.
```
l, r = ui.current.selection()
for ea in range(l, r):
   ...

for ea in database.address():
   ...
```
Or if you want to feed a specific range to your addresses, you can do
```
bounds = database.address.bounds()
for ea in db.address(bounds):
   ...

for ea in db.address(start_ea, end_ea):
   ...
```

**Comment by Harding-Stardust** [comment]
So what I was trying to express was that  you can replace: https://github.com/arizvisa/ida-minsc/blob/af07f167a0992aed3f3a2a1670593880baa57820/misc/ui.py#L299C9-L310C70
with the following code:
```
ok, ea_l, ea_r = ida_kernwin.read_range_selection(None)
```

**Reply by arizvisa** [reply]
There's many ways to transform code (or even regular words) in any and all language.. But, if the semantics are the same why would there be a need to change it?

For your consideration, though, that code was written back in 2018 as commit a921c01af687ef9e42e41e601042742ace994863. This was back before the IDA 7.0 series which introduced `read_range_selection`. The code hasn't had to change because it never stopped working, so when it breaks... it will get updated. This is simply the natural lifecycle of **all** software. Hope this helps clarify the reasoning of why things are the way they are.
